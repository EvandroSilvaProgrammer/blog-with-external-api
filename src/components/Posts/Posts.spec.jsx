import { render, screen } from "@testing-library/react";
import { Posts } from ".";

//Describe the Posts component (MOCK)
const props = {
    posts: [
        {
            id: 1,
            title: "title 1",
            body: "body 1",
            cover: "cover 1",
        },
        {
            id: 2,
            title: "title 2",
            body: "body 2",
            cover: "cover 2",
        },
        {
            id: 3,
            title: "title 3",
            body: "body 3",
            cover: "cover 3",
        }
    ]
};

describe('<Posts />', () => {
    it('should render posts', () => {
        render(<Posts {...props} />);

        expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);

        expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);

        expect(screen.getAllByText(/body/i)).toHaveLength(3);
    });

    it('should match snapshot', () => {
        const { container } = render(<Posts {...props} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
