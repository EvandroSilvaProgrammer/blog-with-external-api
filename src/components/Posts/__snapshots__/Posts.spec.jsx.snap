// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="cover 1"
    />
    <div
      class="post-content"
    >
      <h1>
        title 1
      </h1>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="cover 2"
    />
    <div
      class="post-content"
    >
      <h1>
        title 2
      </h1>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="cover 3"
    />
    <div
      class="post-content"
    >
      <h1>
        title 3
      </h1>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
